lib/internal/cluster/round_robin_handle.js

```diff
@@ -84,32 +98,46 @@ RoundRobinHandle.prototype.remove = function(worker) {
 
 RoundRobinHandle.prototype.distribute = function(err, handle) {
   this.handles.push(handle);
-  const worker = this.free.shift();
-
-  if (worker)
-    this.handoff(worker);
+  this.handoff();
 };
 
-RoundRobinHandle.prototype.handoff = function(worker) {
-  if (worker.id in this.all === false) {
-    return;  // Worker is closing (or has closed) the server.
+RoundRobinHandle.prototype.handoff = function() {
+  const handle = this.handles[0];
+  var socket;
+
+  // There are currently no requests to schedule.
+  if (handle === undefined)
+    return;
+
+  if (this.scheduler.exposeSocket === true) {
+    socket = new net.Socket({
+      handle,
+      readable: false,
+      writable: false,
+      pauseOnCreate: true
+    });
   }
 
-  const handle = this.handles.shift();
+  const worker = this.scheduler(this.workers, socket);
```

Comment:
Should this .slice() or should there be a note in the documentation that the callee is not allowed to modify the workers array?